hedera-node/src/main/java/com/hedera/services/ledger/properties/AccountProperty.java

```diff
@@ -212,5 +217,38 @@
 		public Function<MerkleAccount, Object> getter() {
 			return MerkleAccount::getAlias;
 		}
+	},
+	CRYPTO_ALLOWANCES {
+		@Override
+		public BiConsumer<MerkleAccount, Object> setter() {
+			return (a, t) -> a.setCryptoAllowances(new TreeMap<>((Map<EntityNum, Long>) t));
```

Comment:
Do we really need to wrap `t` in a second new `TreeMap`? 

Afaict we will have always created a new `TreeMap` before ever calling this setter. (And if we didn't, we wouldn't have been able to change the map returned by the `getter()`, since `MerkleAccountState` returns an unmodifiable map.)

Reply:
```
private void setPropsWithSource(final K id, final A entity, final Function<P, Object> extantProps) {
		final var changeSet = changes.get(id);
		for (final var prop : allProps) {
			if (changeSet != null && changeSet.containsKey(prop)) {
				prop.setter().accept(entity, changeSet.get(prop));
			} else {
				prop.setter().accept(entity, extantProps.apply(prop));
			}
		}
	}
```
gets and sets without creating a map. Had to add that wrapper just for this.

Reply:
Ah...got it. Maybe then we could do,
```
return (a, t) -> {
    if (t instanceof TreeMap) {
        a.setCryptoAllowances(t);
    } else {
        a.setCryptoAllowances(new TreeMap<>((Map<EntityNum, Long>) t));
    }
}
```

Reply:
Hmm...I thought more on how this will work with the contract precompile access to accounts; and still, every time a precompile needs an account, it will create 3 new `TreeMap`s plus and 3 new `ImmutableMap`s. 😞 

So I think we should switch these `getter()` and `setter()` methods for the maps to use "unsafe" versions of the getter/setters in `MerkleAccount`, where:
- The unsafe getter doesn't wrap in an `unmodifiableMap`; and,
- The unsafe setter doesn't require a `SortedMap`.